src/main/java/org/cyclonedx/util/deserializer/VulnerabilityDeserializer.java

```diff
@@ -18,34 +18,299 @@
+    if (node.has("cwes")) {
+      JsonNode cwesNode = node.get("cwes");
+      if (cwesNode.isArray()) {
+        List<Integer> codes = mapper.convertValue(node.get("cwes"), new TypeReference<List<Integer>>() {});
+        vulnerability.setCwes(codes);
+      } else if (cwesNode.has("cwe")) {
+        JsonNode cweNode = cwesNode.get("cwe");
+        if (cweNode.isArray()) {
+          List<Integer> codes = mapper.convertValue(cweNode, new TypeReference<List<Integer>>() {
+          });
+          vulnerability.setCwes(codes);
+        } else {
+          vulnerability.setCwes(Collections.singletonList(cweNode.asInt()));
+        }
+      }
+    }
+
+    if (node.has("advisories")) {
+      JsonNode advisoriesNode = node.get("advisories");
+      if (advisoriesNode.isArray()) {
+        List<Vulnerability.Advisory> advisories = mapper.convertValue(node.get("advisories"),
+                new TypeReference<List<Vulnerability.Advisory>>() {});
+        vulnerability.setAdvisories(advisories);
+      } else if (advisoriesNode.has("advisory")) {
+        JsonNode advisoryNode = advisoriesNode.get("advisory");
+        if (advisoryNode.isArray()) {
+          List<Vulnerability.Advisory> advisories = mapper.convertValue(advisoryNode,
+                  new TypeReference<List<Vulnerability.Advisory>>() {});
+          vulnerability.setAdvisories(advisories);
+        } else {
+          vulnerability.setAdvisories(Collections.singletonList(
+                  mapper.convertValue(advisoryNode, Vulnerability.Advisory.class)));
+        }
+      }
+    }
+
+    if (node.has("created")) {
+      vulnerability.setCreated(TimestampUtils.parseTimestamp(node.get("created").textValue()));
+    }
+
+    if (node.has("published")) {
+      vulnerability.setPublished(TimestampUtils.parseTimestamp(node.get("published").textValue()));
+    }
+
+    if (node.has("updated")) {
+      vulnerability.setUpdated(TimestampUtils.parseTimestamp(node.get("updated").textValue()));
+    }
+
+    if (node.has("rejected")) {
+      vulnerability.setRejected(TimestampUtils.parseTimestamp(node.get("rejected").textValue()));
+    }
+
+    if (node.has("credits")) {
+      parseCredits(node.get("credits"), vulnerability, mapper);
+    }
+
+    if (node.has("analysis")) {
+      parseAnalysis(node.get("analysis"), vulnerability, mapper);
+    }
+
+    if (node.has("affects")) {
+      JsonParser affectsParser = node.get("affects").traverse(jsonParser.getCodec());
+      affectsParser.nextToken();
+      List<Vulnerability.Affect> affects = affectsDeserializer.deserialize(affectsParser, ctxt);
+      vulnerability.setAffects(affects);
+    }
+
+    if (node.has("properties")) {
+      JsonParser propertiesParser = node.get("properties").traverse(jsonParser.getCodec());
+      propertiesParser.nextToken();
+      List<Property> properties = propertiesDeserializer.deserialize(propertiesParser, ctxt);
+      vulnerability.setProperties(properties);
+    }
+
+    if (node.has("tools")) {
```

Comment:
what if it's an older version and has only the toolChoice?

Reply:
> what if it's an older version and has only the toolChoice?

Looking at the [docs](https://cyclonedx.org/docs/1.6/json/#vulnerabilities_items_tools), both the new and deprecated field has the same name `Tools`. Also the [xsd](https://github.com/CycloneDX/cyclonedx-core-java/blob/dde00242ae0271e38a0e4312fad0136d801f219c/src/main/resources/bom-1.6.xsd#L173) only defines `Tools` not `Tools` and `ToolChoice`. 

However looking at the docs, the field is of type `Object`, while the previous type was `Array`. For this reason I'm dropping the `JacksonXmlElementWrapper` annotation on the new type.  See 691649c4. Let me know if you think it should be added back.

Reply:
Get it, this if looks the same as the one in the metadata deserializer, I would have one method for both in a util classes or something

Reply:
> Get it, this if looks the same as the one in the metadata deserializer, I would have one method for both in a util classes or something

28e15c1